From 4ae0af9c9a29c919f110742d4e6085efad581870 Mon Sep 17 00:00:00 2001
From: Chuanxiao Dong <chuanxiao.dong@intel.com>
Date: Mon, 9 Mar 2015 11:03:45 +0800
Subject: [PATCH] mmc: host: disable the power resource for the SD host

Power resource defined in DSDT can shut down the SD card's power
when system is still in S0. So driver has to by pass the power
resource management in ACPI

Change-Id: Iba14d56ae6b10ebd6607f0299fe79ff5211ca4d8
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-24795
Signed-off-by: Chuanxiao Dong <chuanxiao.dong@intel.com>
---
 drivers/mmc/host/sdhci-acpi.c | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/drivers/mmc/host/sdhci-acpi.c b/drivers/mmc/host/sdhci-acpi.c
index 2a02966..6bf1c70 100644
--- a/drivers/mmc/host/sdhci-acpi.c
+++ b/drivers/mmc/host/sdhci-acpi.c
@@ -158,6 +158,18 @@ static int sdhci_acpi_sdio_probe_slot(struct platform_device *pdev)
 	return sdhci_acpi_probe_slot(pdev);
 }
 
+static int sdhci_acpi_sd_probe_slot(struct platform_device *pdev)
+{
+	struct device *dev = &pdev->dev;
+	acpi_handle handle = ACPI_HANDLE(dev);
+	struct acpi_device *device;
+
+	if (!acpi_bus_get_device(handle, &device))
+		device->power.flags.power_resources = 0;
+
+	return sdhci_acpi_probe_slot(pdev);
+}
+
 static int sdhci_acpi_remove_slot(struct platform_device *pdev)
 {
 	struct sdhci_acpi_host *c = platform_get_drvdata(pdev);
@@ -202,7 +214,7 @@ static const struct sdhci_acpi_slot sdhci_acpi_slot_int_sdio = {
 static const struct sdhci_acpi_slot sdhci_acpi_slot_int_sd = {
 	.flags   = SDHCI_ACPI_SD_CD | SDHCI_ACPI_RUNTIME_PM,
 	.quirks2 = SDHCI_QUIRK2_PRESET_VALUE_BROKEN,
-	.probe_slot = sdhci_acpi_probe_slot,
+	.probe_slot = sdhci_acpi_sd_probe_slot,
 	.remove_slot = sdhci_acpi_remove_slot,
 };
 
-- 
1.9.1

